docker-app/qfieldcloud/core/models.py

```diff
@@ -691,6 +698,38 @@ def active_users(self, period_since: datetime, period_until: datetime):
             is_staff=False,
         ).filter(Q(id__in=users_with_delta) | Q(id__in=users_with_jobs))
 
+    def active_users_jobs_deltas_count(
+        self, period_since: datetime, period_until: datetime
+    ) -> List[Person]:
+        """Returns the list of active users annotated with jobs_count and deltas_count."""
+        active_users = self.active_users(period_since, period_until)
+
+        jobs_created: QuerySet = (
+            self.jobs(period_since, period_until)
+            .values("created_by")
+            .annotate(jobs_count=Count("created_by"))
+        )
+        deltas_created: QuerySet = (
+            self.deltas(period_since, period_until)
+            .values("created_by")
+            .annotate(deltas_count=Count("created_by"))
+        )
+        for user in active_users:
+            # NOTE calling qs.first() is wrongly returning 1 instead of the count value
+            qs = jobs_created.filter(created_by=user)
+            user.jobs_count = qs.values_list("jobs_count", flat=True)[0] if qs else 0
+
+            qs = deltas_created.filter(created_by=user)
+            user.deltas_count = (
+                qs.values_list("deltas_count", flat=True)[0] if qs else 0
+            )
+
+        return sorted(
+            list(active_users),
+            key=lambda obj: (obj.jobs_count, obj.deltas_count),
+            reverse=True,
+        )
+
```

Comment:
There is sure potential to optimize the queries, currently is doing 45 queries for 2 active users, 41 queries for 1 active user. So it might get slow with many active users ... .  Probably with some prefetch_related magic could be improved. Question is how much more effort want to put into optimizing this.

![image](https://github.com/opengisch/qfieldcloud/assets/21113500/5b611ba7-0655-445d-a26b-f0b051af3579)